internal/scalibrextract/filesystem/vendored/vendored.go

```diff
@@ -0,0 +1,179 @@
+		"thirdparty":  {},
+		"third-party": {},
+		"third_party": {},
+		"libs":        {},
+		"external":    {},
+		"externals":   {},
+		"vendor":      {},
+		"vendored":    {},
+	}
+)
+
+const (
+	// This value may need to be tweaked, or be provided as a configurable flag.
+	determineVersionThreshold = 0.15
+	maxDetermineVersionFiles  = 10000
+)
+
+type Extractor struct {
+	// ScanGitDir determines whether a vendored library with a git directory is scanned or not,
+	// this is used to avoid duplicate results, once from git scanning, once from vendoredDir scanning
+	ScanGitDir bool
+	// TODO(v2): Client rework
+	// determineVersionsClient
+}
+
+var _ filesystem.Extractor = Extractor{}
+
+// Name of the extractor.
+func (e Extractor) Name() string { return "filesystem/vendored" }
+
+// Version of the extractor.
+func (e Extractor) Version() int { return 0 }
+
+// Requirements of the extractor.
+func (e Extractor) Requirements() *plugin.Capabilities {
+	return &plugin.Capabilities{}
+}
+
+// FileRequired returns true for .package-lock.json files under node_modules
+func (e Extractor) FileRequired(fapi filesystem.FileAPI) bool {
+	// Check if parent directory is one of the vendoredLibName
+	// Clean first before Dir call to avoid trailing slashes causing problems
+	parentDir := filepath.Base(filepath.Dir(filepath.Clean(fapi.Path())))
+	_, ok := vendoredLibNames[parentDir]
+	if !ok {
+		return false
+	}
+
+	// Stat costs performance, so perform it after the name check
+	stat, err := fapi.Stat()
+	if err != nil {
+		return false
+	}
+
+	return stat.IsDir()
+}
+
+// Extract extracts packages from yarn.lock files passed through the scan input.
+func (e Extractor) Extract(_ context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
+	var packages []*extractor.Inventory
+
+	// r.Infof("Scanning potential vendored dir: %s\n", libPath)
+	// TODO: make this a goroutine to parallelize this operation
+	results, err := queryDetermineVersions(input.Path, input.FS, e.ScanGitDir)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(results.Matches) > 0 && results.Matches[0].Score > determineVersionThreshold {
+		match := results.Matches[0]
+		// r.Infof("Identified %s as %s at %s.\n", libPath, match.RepoInfo.Address, match.RepoInfo.Commit)
+		packages = append(packages, &extractor.Inventory{
+			SourceCode: &extractor.SourceCodeIdentifier{
+				Commit: match.RepoInfo.Commit,
+			},
+			Locations: []string{input.Path},
+		})
+	}
+
+	return packages, nil
+}
+
+// ToPURL converts an inventory created by this extractor into a PURL.
+func (e Extractor) ToPURL(_ *extractor.Inventory) *purl.PackageURL {
+	return nil
+}
+
+// Ecosystem returns the OSV ecosystem ('npm') of the software extracted by this extractor.
+func (e Extractor) Ecosystem(_ *extractor.Inventory) string {
+	return ""
+}
+
+func queryDetermineVersions(repoDir string, fsys scalibrfs.FS, scanGitDir bool) (*osv.DetermineVersionResponse, error) {
+	fileExts := []string{
```

Comment:
should we move this up as a global instead to be consistent with `vendoredLibNames` ?

Reply:
Yep, done